log4j-core/src/main/java/org/apache/logging/log4j/core/pattern/MapPatternConverter.java

```diff
@@ -60,9 +60,9 @@ public static MapPatternConverter newInstance(final String[] options) {
      */
     @Override
     public void format(final LogEvent event, final StringBuilder toAppendTo) {
-        StringMapMessage msg;
-        if (event.getMessage() instanceof StringMapMessage) {
-            msg = (StringMapMessage) event.getMessage();
+        MapMessage msg;
+        if (event.getMessage() instanceof MapMessage) {
+            msg = (MapMessage) event.getMessage();
```

Comment:
Unfortunately this doesn't quite work because we cannot rely on casting map values to strings on line 82

Reply:
just remove the cast?